lib/file-view.coffee

```diff
@@ -19,7 +19,11 @@ class FileView extends HTMLElement
     @fileName.dataset.name = @file.name
     @fileName.dataset.path = @file.path
 
-    @fileName.classList.add(FileIcons.getService().iconClassForPath(@file.path))
+    iconClass = FileIcons.getService().iconClassForPath(@file.path, this)
```

Comment:
@as-cii Okay, it's done! :D You'll notice I've passed a [`@stats` property](https://github.com/atom/tree-view/pull/825/files#diff-fe4fb3eaf876e081a80d3f01512c120aR72) from the newly-added `File` method, which will be smoothly taken care of by #859.

The metadata includes atomic values with no dangling references. I only included stuff that was strictly filesystem-related, and not in any way related to the tree-view itself (which I believe was your point in abstracting the consumer's logic from the provider's).